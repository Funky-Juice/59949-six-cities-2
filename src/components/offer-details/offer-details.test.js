import renderer from 'react-test-renderer';
import reviewTestObj from '../../mocks/test-review';
import offerTestObj from '../../mocks/test-offer';
import OfferDetails from './offer-details';
import {createStore} from 'redux';
import {Provider} from 'react-redux';

const mockReviews = [reviewTestObj];
const mockOffer = offerTestObj;

const store = createStore(() => ({
  reviews: mockReviews,
  activeCity: mockOffer.city,
  activeOffers: [mockOffer],
  isAuthorizationRequired: false
}));

it(`OfferDetails correctly renders`, () => {
  const tree = renderer
    .create(<Provider store={store}>
      <OfferDetails
        offer={mockOffer}
      />
    </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
