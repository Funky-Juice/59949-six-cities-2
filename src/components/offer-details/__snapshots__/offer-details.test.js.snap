// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferDetails correctly renders 1`] = `
<section
  className="property"
>
  <div
    className="property__gallery-container container"
  >
    <div
      className="property__gallery"
    >
      <div
        className="property__image-wrapper"
      >
        <img
          alt="img"
          className="property__image"
          src="room.jpg"
        />
      </div>
      <div
        className="property__image-wrapper"
      >
        <img
          alt="img"
          className="property__image"
          src="room.jpg"
        />
      </div>
      <div
        className="property__image-wrapper"
      >
        <img
          alt="img"
          className="property__image"
          src="room.jpg"
        />
      </div>
      <div
        className="property__image-wrapper"
      >
        <img
          alt="img"
          className="property__image"
          src="room.jpg"
        />
      </div>
      <div
        className="property__image-wrapper"
      >
        <img
          alt="img"
          className="property__image"
          src="room.jpg"
        />
      </div>
      <div
        className="property__image-wrapper"
      >
        <img
          alt="img"
          className="property__image"
          src="room.jpg"
        />
      </div>
    </div>
  </div>
  <div
    className="property__container container"
  >
    <div
      className="property__wrapper"
    >
      <div
        className="property__name-wrapper"
      >
        <h1
          className="property__name"
        >
          Wood and stone place
        </h1>
        <button
          className="
          button
          property__bookmark-button
          property__bookmark-button--active
        "
          onClick={[Function]}
          type="button"
        >
          <svg
            className="property__bookmark-icon"
            height="33"
            width="31"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="property__rating rating"
      >
        <div
          className="property__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
        <span
          className="property__rating-value rating__value"
        >
          3
        </span>
      </div>
      <ul
        className="property__features"
      >
        <li
          className="property__feature property__feature--bedrooms"
        >
           
          Bedrooms
        </li>
        <li
          className="property__feature property__feature--adults"
        >
          Max 
           
          adults
        </li>
      </ul>
      <div
        className="property__price"
      >
        <b
          className="property__price-value"
        >
          €
          80
        </b>
        <span
          className="property__price-text"
        >
           night
        </span>
      </div>
      <div
        className="property__inside"
      >
        <h2
          className="property__inside-title"
        >
          What's inside
        </h2>
        <ul
          className="property__inside-list"
        >
          <li
            className="property__inside-item"
          >
            Wi-Fi
          </li>
        </ul>
      </div>
      <div
        className="property__host"
      >
        <h2
          className="property__host-title"
        >
          Meet the host
        </h2>
        <div
          className="property__host-user user"
        >
          <div
            className="property__avatar-wrapper  user__avatar-wrapper"
          >
            <img
              alt="Host avatar"
              className="property__avatar user__avatar"
              height="74"
              src="/img/avatar.svg"
              width="74"
            />
          </div>
          <span
            className="property__user-name"
          >
            Max
          </span>
          <span
            className="property__user-status"
          />
        </div>
        <div
          className="property__description"
        />
      </div>
      <section
        className="property__reviews reviews"
      >
        <h2
          className="reviews__title"
        >
          Reviews
          <span
            className="reviews__amount"
          >
             ·
            1
          </span>
        </h2>
        <ul
          className="reviews__list"
        >
          <li
            className="reviews__item"
          >
            <div
              className="reviews__user user"
            >
              <div
                className="reviews__avatar-wrapper user__avatar-wrapper"
              >
                <img
                  alt="Reviews avatar"
                  className="reviews__avatar user__avatar"
                  height="54"
                  src="https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/5.jpg"
                  width="54"
                />
              </div>
              <span
                className="reviews__user-name"
              >
                Oliver
              </span>
            </div>
            <div
              className="reviews__info"
            >
              <div
                className="reviews__rating rating"
              >
                <div
                  className="reviews__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "60%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <p
                className="reviews__text"
              >
                random text
              </p>
              <time
                className="reviews__time"
                dateTime="2019-04-24"
              >
                December
                 
                2019
              </time>
            </div>
          </li>
        </ul>
        <form
          action="#"
          className="reviews__form form"
          method="post"
          onSubmit={[Function]}
        >
          <label
            className="reviews__label form__label"
            htmlFor="review"
          >
            Your review
          </label>
          <div
            className="reviews__rating-form form__rating"
          >
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              id="5-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={5}
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="5-stars"
              title="perfect"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              id="4-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={4}
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="4-stars"
              title="good"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              id="3-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={3}
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="3-stars"
              title="not bad"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              id="2-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={2}
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="2-stars"
              title="badly"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              id="1-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={1}
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="1-stars"
              title="terribly"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
          </div>
          <textarea
            className="reviews__textarea form__textarea"
            id="review"
            name="review"
            onChange={[Function]}
            placeholder="Tell how was your stay, what you like and what can be improved"
            value=""
          />
          <div
            className="reviews__button-wrapper"
          >
            <p
              className="reviews__help"
            >
              To submit review please make sure to set 
              <span
                className="reviews__star"
              >
                rating
              </span>
              and describe your stay with at least
              <b
                className="reviews__text-amount"
              >
                 50 characters
              </b>
              .
            </p>
            <button
              className="reviews__submit form__submit button"
              disabled={true}
              type="submit"
            >
              Submit
            </button>
          </div>
          <div
            style={
              Object {
                "color": "red",
                "fontSize": "14px",
                "position": "relative",
              }
            }
          >
            <p
              className="reviews__error-message"
              style={
                Object {
                  "position": "absolute",
                }
              }
            />
          </div>
        </form>
      </section>
    </div>
  </div>
  <section
    className="property__map map"
  >
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  </section>
</section>
`;
